### DISCUSSION TITLE
API testing plan for Stacks 2.1

### DISCUSSION THREAD

**Discussion by zone117x** [blockstack/stacks-blockchain-api#1257] [General]
Discussion to track known changes that the API needs to handle Stacks 2.1, and the testing required.

WIP list:
* Determine any required changes for new PoX locking/unlock event handling, e.g. https://github.com/stacks-network/stacks-blockchain/issues/2534 and https://github.com/stacks-network/stacks-blockchain/issues/2533
* Rosetta uses a hardcoded pox contract (`SP000000000000000000002Q6VF78.pox`) which needs updated IIUC.
* Support reading `memo` from stx-transfer events, see https://github.com/stacks-network/stacks-blockchain/pull/2488
* Support for parsing new coinbase transaction payloads, see https://github.com/stacks-network/stacks-blockchain/pull/3164
* Support for parsing new smart-contract transaction payloads, see https://github.com/stacks-network/stacks-blockchain/pull/3192

**Comment by rafaelcr** [comment]
~~Something that also comes to mind is all the BNS stuff (endpoints, tables, imports, etc.). We'll need to turn that off and/or gate it behind a run mode or something similar.~~

EDIT: I got confused with Hyperchains support 😓
